File: 2025/regional-andaluza/A/code/cpp/A.cpp
#include <iostream>
using namespace std;

bool es_potencia_de_2(int n) {
    return (n & (n - 1)) == 0;
}

int contar_combates(int n) {
    int combates = 0;
    while (n > 1) {
        if (es_potencia_de_2(n)) {
            combates += n - 1;
            n = 1;
        } else {
            int c = n / 2;
            combates += c;
            n = c;
            if (c % 2 == 1 && c != 1) {
                n += 1; 
            }
        }
    }
    return combates;
}

int main() {
    int T, N;
    cin >> T;
    for (int i = 0; i < T; ++i) {
        cin >> N;
        cout << contar_combates(N) << endl;
    }
    return 0;
}

File: 2025/regional-andaluza/A/code/py/A.py
def es_potencia_de_2(n):
    return (n & (n - 1)) == 0

def contar_combates(n):
    combates = 0
    while n > 1:
        if es_potencia_de_2(n):
            combates += n - 1
            n = 1  
        else:
            combates += n // 2
            n = n // 2
            if n % 2 == 1 and n != 1:
                n += 1
    return combates

if __name__ == "__main__":
    T = int(input())
    for _ in range(T):
        N = int(input())
        print(contar_combates(N))
